cabal-install/src/Distribution/Client/Win32SelfUpgrade.hs

```diff
@@ -140,7 +140,7 @@ deleteOldExeFile verbosity oldPID tmpPath = do
      ++ show oldPID ++ " at path " ++ tmpPath
 
   log $ "getting handle of parent process " ++ show oldPID
-  oldPHANDLE <- Win32.openProcess Win32.sYNCHORNIZE False (fromIntegral oldPID)
```

Comment:
This seems to be an uncaught bug.

Reply:
I think this was caught and fixed in Win32 long ago, but then not fixed here.